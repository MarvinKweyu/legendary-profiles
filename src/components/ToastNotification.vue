<template>
  <div
    class="my-notification bg-danger position-fixed bottom-0 end-0 p-3 mb-1"
    role="alert"
    aria-live="assertive"
    aria-atomic="true"
    style="z-index: 5"
    data-bs-autohide="false"
    data-bs-delay="10000"
  >
    <div class="toast-header">
      <strong class="me-auto">Unable to get user</strong>
      <button
        type="button"
        class="btn-close"
        data-bs-dismiss="toast"
        aria-label="Close"
      ></button>
    </div>
    <div class="toast-body">
      Unable to get the specified user: {{ errorDetail.userSearch }}
    </div>
  </div>
</template>

<script>
export default {
  name: "ToastNotification",
  props: {
    errorDetail: {
      type: Object,
      required: true,
      default: () => ({}),
    },
  },
};
</script>

<style scoped>
.my-notification {
  border-radius: 10px !important;
}
</style>
