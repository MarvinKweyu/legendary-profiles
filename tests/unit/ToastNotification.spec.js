import {shallowMount, createLocalVue } from "@vue/test-utils"
import Vuex from 'vuex'
import initialState from '@/store/state'
import ToastNotification from '@/components/ToastNotification'
import errorDetail from './fixtures/error'

const localVue = createLocalVue()
localVue.use(Vuex)

describe("ToastNotification", () =>{

    let props
    const build = () =>{
        const wrapper =  shallowMount(ToastNotification, {
            propsData: props
        })

        return {
            wrapper,
            toastHeader: () => wrapper.find('.toast-header'),
            toastBody: () => wrapper.find('.toast-body'),
            toastCloseBtn: () => wrapper.find('.btn-close')
        }
    }

    beforeEach(() =>{
         props = {
            errorDetail
        }
    })

    it("renders the component", () =>{
        const { wrapper }=  build()
        expect(wrapper.html()).toMatchSnapshot();
    })

    it("renders main sections of the component", () =>{
        const {toastHeader, toastBody, toastCloseBtn} = build()

        expect(toastHeader().exists()).toBe(true)

        expect(toastBody().exists()).toBe(true)
        expect(toastBody().text()).toContain(props.errorDetail.userSearch)

        expect(toastCloseBtn().exists()).toBe(true)

    })
})