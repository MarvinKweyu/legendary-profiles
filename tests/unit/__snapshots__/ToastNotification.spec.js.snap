// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastNotification renders the component 1`] = `
<div role="alert" aria-live="assertive" aria-atomic="true" data-bs-autohide="false" data-bs-delay="10000" class="toast bg-danger position-fixed bottom-0 end-0 p-3" style="z-index: 5;">
  <div class="toast-header"><strong class="me-auto">Unable to get user</strong> <small>11 mins ago</small> <button type="button" data-bs-dismiss="toast" aria-label="Close" class="btn-close"></button></div>
  <div class="toast-body">
    Unable to get the specified user: IDoNotExist
  </div>
</div>
`;
